### ISSUE TITLE
parallelizing code on multicore CPU

### ISSUE THREAD

**Issue by DanPuzzuoli** [google/jax#2714]
Hi,

I have a vague newby question: Should I expect `jit`ed functions to automatically make use of multiple cores in a CPU (unfortunately my GPU does not support CUDA)? 

The context that I'm asking is that I have a function in `jax` that I'm optimizing using one of the gradient methods in `scipy.optimize.minimize`. Things are working really well (I like coding in jax and the autodiff is great), but only one core is being used. There is a portion of the function (in fact the most expensive part) that is highly parallelizable. 

Should compiled functions automatically parallelize and make use of the cores of a device, or do I need to do something to parallelize that step of the computation? I get that I may need to write things in a better way so that the compilation "recognizes" that that step can be parallelized, but I guess I'm generally asking if I write things "correctly" should the parallelization be automatic? Or is it necessary to explicitly parallelize steps? (I see that there are parallelization tools in `jax` but those functions seem to be more about running the same code in parallel on multiple devices, not about utilizing cores of a single device.)

Apologies for the newb question! I'm new to the world of auto-differentiation and XLA, but I'm really loving using JAX and want to utilize it correctly!

**Comment by froystig** [comment]
You can manually parallelize on CPU by setting the XLA flag `xla_force_host_platform_device_count` (see #1408) and using `jax.pmap` (see [documentation](https://jax.readthedocs.io/en/latest/jax.html#parallelization-pmap)).

Separately, note that intra-op parallelism might make use of multiple cores. That is, whether under `jit` or not, XLA (and therefore JAX) may choose to run some operations, such as matrix multiplication, on multiple CPU cores.

Please reopen this issue if it doesn't fully address your question!

**Comment by ftgreat** [comment]
https://bnikolic.co.uk/blog/python/jax/2023/03/22/jax-multithreaded.html
